Tests/Engine/Results/BacktestingResultHandlerTests.cs

```diff
@@ -0,0 +1,391 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Deedle;
+using NUnit.Framework;
+using QuantConnect.Algorithm.CSharp;
+using QuantConnect.Lean.Engine.Results;
+using QuantConnect.Report;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Tests.Engine.Results
+{
+    [TestFixture]
+    public class BacktestingResultHandlerTests
+    {
+        private Dictionary<string, BacktestingResultHandler> _resultsCache = new Dictionary<string, BacktestingResultHandler>();
```

Comment:
If we are just consuming the `Charts`, I'd suggest for us to just store them instead, if we need more in the future we can add it too